scripts/common/kubernetes.sh

```diff
@@ -72,30 +76,57 @@ function kubernetes_install_host_packages() {
         kubernetes_get_host_packages_online "$k8sVersion"
     fi
 
+    cat > "$DIR/tmp-kubeadm.conf" <<EOF
+# Note: This dropin only works with kubeadm and kubelet v1.11+
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
+Environment="KUBELET_CONFIG_ARGS=--config=/var/lib/kubelet/config.yaml"
+# This is a file that "kubeadm init" and "kubeadm join" generates at runtime, populating the KUBELET_KUBEADM_ARGS variable dynamically
+EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
+# This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
+# the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
+EnvironmentFile=-/etc/__ENV_LOCATION__/kubelet
+ExecStart=
+ExecStart=/usr/bin/kubelet \$KUBELET_KUBECONFIG_ARGS \$KUBELET_CONFIG_ARGS \$KUBELET_KUBEADM_ARGS \$KUBELET_EXTRA_ARGS
+EOF
+
     case "$LSB_DIST" in
         ubuntu)
+            sed "s:__ENV_LOCATION__:default:g" -i "$DIR/tmp-kubeadm.conf"
             export DEBIAN_FRONTEND=noninteractive
             dpkg --install --force-depends-version $DIR/packages/kubernetes/${k8sVersion}/ubuntu-${DIST_VERSION}/*.deb
             ;;
 
         centos|rhel|amzn)
             case "$LSB_DIST$DIST_VERSION_MAJOR" in
                 rhel8|centos8)
+                    sed "s:__ENV_LOCATION__:sysconfig:g" -i "$DIR/tmp-kubeadm.conf"
                     rpm --upgrade --force --nodeps $DIR/packages/kubernetes/${k8sVersion}/rhel-8/*.rpm
                     ;;
 
                 *)
+                    sed "s:__ENV_LOCATION__:sysconfig:g" -i "$DIR/tmp-kubeadm.conf"
                     rpm --upgrade --force --nodeps $DIR/packages/kubernetes/${k8sVersion}/rhel-7/*.rpm
                     ;;
             esac
         ;;
     esac
 
+    # Install Kubeadm from binary (see kubernetes.io)
+    cp -f "$DIR/packages/kubernetes/${k8sVersion}/assets/kubeadm" /usr/bin/
+    chmod a+rx /usr/bin/kubeadm
+
+    mkdir -p /etc/systemd/system/kubelet.service.d
+    cp -f "$DIR/tmp-kubeadm.conf" /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+    chmod 755 /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
```

Comment:
I've got 0640 on my test install. I don't think it needs to be executable.

Reply:
I think was an artifact of troubleshooting. I get 644 when I create a test file doing something similar so I'll make sure to change it to 640